src/fault_injector/fault_types/train_prio_fault.py

```diff
@@ -1,27 +1,67 @@
+from src.fault_injector.fault_configurations.train_prio_fault_configuration import (
+    TrainPrioFaultConfiguration,
+)
 from src.fault_injector.fault_types.fault import Fault
+from src.interlocking_component.route_controller import IInterlockingDisruptor
+from src.logger.logger import Logger
+from src.wrapper.simulation_object_updating_component import (
+    SimulationObjectUpdatingComponent,
+)
+from src.wrapper.simulation_objects import Train
 
 
 class TrainPrioFault(Fault):
     """A fault affecting the priority of trains."""
 
+    configuration: TrainPrioFaultConfiguration
+    old_prio: int
+    train: Train
+    wrapper: SimulationObjectUpdatingComponent
+    interlocking: IInterlockingDisruptor
+
+    # pylint: disable=duplicate-code
+    # Otherwise another inheritance layer would be needed. This will be refactored in the future
+    def __init__(
+        self,
+        configuration,
+        logger: Logger,
+        wrapper: SimulationObjectUpdatingComponent,
+        interlocking: IInterlockingDisruptor,
+    ):
+        super().__init__(configuration, logger)
+        self.wrapper = wrapper
+        self.interlocking = interlocking
+
     def inject_fault(self, tick: int):
         """inject TrainPrioFault into the given component
 
         :param tick: the simulation tick in which inject_fault was called
         :type tick: Integer
         """
-        # - get train object
-        # - save the current prio of the train in old_prio
-        # - set train prio to new_prio
-        raise NotImplementedError()
+        self.train: Train = [
```

Comment:
Are you entirely sure that you will find the train? Maybe you want to check for it first.

Reply:
Ah that is a good point 👇. Since I've missed that in the other Faults as well I would fix that with an extra issue.